sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -64,6 +67,8 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
     )
     while True:
         try:
+            client_timeout = kwargs.get('timeout')
+            start_time = time.time()
```

Comment:
Should this be outside the loop?

Reply:
Thanks @johanste 
I don't think so - we calculate the time spent for each loop iteration and deduct it from the absolute timeout, which we then feed back into kwargs on line 122 - so I think it's correct to do this with each iteration